index.js

```diff
@@ -1,29 +1,126 @@
-import actions from '@actions/core';
-import fs from 'fs/promises';
+import coreActions from '@actions/core';
+import githubActions from '@actions/github';
 import fetch from 'node-fetch';
 import fm from 'yaml-front-matter';
 
 // See: https://docs.modrinth.com/#tag/projects/operation/modifyProject
 
+const getGithubRawUrl = async (token) => {
+    const context = githubActions.context;
+    let { owner, repo } = context.repo;
+
+    let branch;
+
+    // 1. Handle Pull Requests
+    if (context.ref && context.ref.startsWith('refs/pull/')) {
+        const pr = context.payload.pull_request;
+        if (!pr) {
+            throw new Error('Pull request data not available in context');
+        }
+
+        // Assuming `pr.head.ref` contains just the branch name (e.g., "feature-branch")
+        branch = pr.head.ref;
+
+        coreActions.info(`PR detected: branch = ${branch}, repo = ${owner}/${repo}`);
+    }
+    // 2. Handle branch pushes
+    else if (context.ref && context.ref.startsWith('refs/heads/')) {
+        branch = context.ref.replace('refs/heads/', '');
+        coreActions.info(`Branch push: ${branch}`);
+    }
+    // 3. Handle tags
+    else if (context.ref && context.ref.startsWith('refs/tags/')) {
+        branch = context.ref.replace('refs/tags/', '');
+        coreActions.info(`Tag: ${branch}`);
+    }
+    // 4. Fallback to default branch
+    else {
+        if (!token) {
+            branch = 'main';
+            coreActions.warning(
+                `Cannot determine branch from ref "${context.ref}" and no repo-token provided. ` +
+                `Defaulting to "main". This may cause file resolution issues.`
+            );
+        } else {
+            const octokit = githubActions.getOctokit(token);
+            try {
+                const { data: repoData } = await octokit.rest.repos.get({ owner, repo });
+                branch = repoData.default_branch;
+                coreActions.info(`Using default branch: ${branch}`);
+            } catch (error) {
+                throw new Error(`Failed to get repository info: ${error.message}`);
+            }
+        }
+    }
```

Comment:
I'm okay with just passing the branch as an action parameter (defaulting to `main`), rather than trying to figure it out using this method.